addons/project_timesheet_holidays/models/account_analytic.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticLine(models.Model):
+
+    _inherit = 'account.analytic.line'
+
+    holiday_id = fields.Many2one("hr.holidays", string='Leave Request')
+
+    @api.multi
+    def unlink(self):
+        if self.mapped('holiday_id'):
```

Comment:
Shouln't you check the holidays state ? You ask for canceling, but it will raise again.